runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearsOfGuthixPlugin.java

```diff
@@ -112,4 +132,27 @@ public void onDecorativeObjectDespawned(DecorativeObjectDespawned event)
 		DecorativeObject object = event.getDecorativeObject();
 		streams.remove(object);
 	}
+
+	Skill getTearsOfGuthixSkill()
+	{
+		final int[] skillXps = client.getSkillExperiences();
+		final Skill[] skills = Skill.values();
+
+		Skill skill = skills[0];
+		int lowestXp = skillXps[0];
+
+		// Ignore overall skill experience (last value)
+		for (int i = 1; i < skills.length - 1; i++)
+		{
+			final int skillXp = skillXps[i];
+			final Skill curSkill = skills[i];
+			if (skillXp < lowestXp || (skillXp == lowestXp && TOG_SKILLS.indexOf(skill) > TOG_SKILLS.indexOf(curSkill)))
+			{
+				skill = curSkill;
+				lowestXp = skillXp;
+			}
+		}
+
+		return skill;

```

Comment:
Instead of looping over `Skill`, use `client.getSkillExperience(Skill skill)` and only loop over `TOG_SKILLS`. This has the added benefit of removing the index check since it iterates over `TOG_SKILLS` starting from the lowest index.

```suggestion
		Skill skill = TOG_SKILLS.get(0);
		int lowestXp = client.getSkillExperience(skill);

		for (int i = 1; i < TOG_SKILLS.size(); i++)
		{
			final Skill curSkill = TOG_SKILLS.get(i);
			final int skillXp = client.getSkillExperience(curSkill);

			if (skillXp < lowestXp)
			{
				skill = curSkill;
				lowestXp = skillXp;
			}
		}

		return skill;
```